cocinaborcelle/utils/userContext.tsx

```diff
@@ -0,0 +1,54 @@
+import axios from "axios";
+import { User } from "lucia";
+import { useRouter } from "next/router";
+import {
+	ReactNode,
+	createContext,
+	useContext,
+	useEffect,
+	useState,
+} from "react";
+
+const UserContext = createContext<User | null>(null);
+
+export function UserProvider({ children }: { children: ReactNode }) {
+	let authNeedPages = ["/user"];
+
+	const router = useRouter();
+	const [usuario, setUsuario] = useState<null | User>(null);
+
+	useEffect(() => {
+		axios
+			.get("/api/auth/getUser")
+			.then((response) => {
+				console.log(
+					router.pathname,
+					authNeedPages.includes(router.pathname)
+				);
+				if (authNeedPages.includes(router.pathname)) {
+					if (response.data?.user) {
+						setUsuario(response.data.user);
+					} else {
+						router.replace("/login");
+					}
+				} else {
+					if (response.data?.user) {
+						setUsuario(response.data.user);
+					} else {
+						setUsuario(null);
+					}
+				}
```

Comment:
```suggestion
				if (pagesWithoutAuth.includes(router.pathname)) {
                                    if (response.data?.user) {
                                        setUsuario(response.data.user);
				    } else {
                                        setUsuario(null);
				    }
                                } else {
                                    if (response.data?.user) {
                                        setUsuario(response.data.user);
                                    } else {
                                        router.replace("/login");
                                    }
				}
```

Nessas questões de permissão de usuário e de acesso a partes da aplicação, acho melhor utilizar a lógica inversa. Em vez de pensar "qual página precisa de autenticação?", pensar "qual página não precisa de autenticação?", porque, caso alguém faça uma nova página com dados sensíveis, é melhor ela automaticamente precisar de autenticação do que confiar no desenvolvedor em adicionar essa página neste arquivo